File: apps/windows/TeslaSync.App/feature-views/WebhooksPage/WebhooksPage.Model.cs
// Notifications / Webhooks page — native model layer.
//
// The WinUI-free registration, render-model and diagnostics behind the
// WebhooksPage surface (the native parity port of the web page
// web/src/features/notifications/pages/WebhooksPage.tsx). The web page is a thin
// PageContainer (localized title + subtitle + a "copy link" action) wrapping the
// shared WebhookChannelsSection; this page mirrors that exactly — the page chrome
// lives here, the list/CRUD lives in the already-ported WebhookChannelsSection.
//
// Everything here is UI-thread-free so the registration metadata, the localized
// title/subtitle projection and the PII-safe diagnostics are asserted headlessly
// without a WinUI host.
using System.Threading;
using TeslaSync.App.Core.Notifications;

namespace TeslaSync.App.FeatureViews.Notifications;

/// <summary>
/// Canonical metadata for the Webhooks page — the native mirror of the web route <c>/notifications/webhooks</c>
/// (nav name <c>Webhooks</c>). The shell page factory registers the surface under <see cref="RouteName"/>; the
/// title and subtitle resolve through the i18n facade with the web key names (the inline <c>t()</c> defaults from
/// <c>WebhooksPage.tsx</c>).
/// </summary>
public static class WebhooksPageRegistration
{
    /// <summary>The navigation route name the shell page factory registers this surface under.</summary>
    public const string RouteName = "NotificationsWebhooks";

    /// <summary>The normalized route path (web route <c>/notifications/webhooks</c>) used to build the share link.</summary>
    public const string RoutePath = "notifications/webhooks";

    /// <summary>Diagnostics surface slug emitted with the <c>view.opened</c> event.</summary>
    public const string Slug = "WebhooksPage";

    /// <summary>The web page this surface mirrors.</summary>
    public const string WebSource = "features/notifications/pages/WebhooksPage.tsx";

    /// <summary>The localized page title (web <c>notifications.webhooks.title</c>).</summary>
    public static string Title(ILocalizer localizer)
    {
        ArgumentNullException.ThrowIfNull(localizer);
        return localizer.GetString("notifications.webhooks.title", "Webhooks");
    }

    /// <summary>The localized page subtitle (web <c>notifications.webhooks.subtitle</c>).</summary>
    public static string Subtitle(ILocalizer localizer)
    {
        ArgumentNullException.ThrowIfNull(localizer);
        return localizer.GetString(
            "notifications.webhooks.subtitle",
            "Custom HTTPS endpoints that receive HMAC-signed event payloads.");
    }
}

/// <summary>
/// The render-ready projection the <see cref="WebhooksPageViewModel"/> exposes for the page chrome — the localized
/// <see cref="Title"/> and <see cref="Subtitle"/> (web <c>PageContainer</c> header) plus the page's accessible
/// <see cref="AutomationName"/>. The webhook list and its loading / empty / error states are owned by the embedded
/// <see cref="WebhookChannelsSection"/>, so the page itself carries no data state of its own.
/// </summary>
/// <param name="Title">The localized page title (web <c>PageContainer title</c>).</param>
/// <param name="Subtitle">The localized page subtitle (web <c>PageContainer subtitle</c>).</param>
/// <param name="AutomationName">The Narrator name announced for the page region.</param>
public sealed record WebhooksDisplay(string Title, string Subtitle, string AutomationName);

/// <summary>
/// PII-safe diagnostics for the Webhooks page (P1/S11 diagnostics contract). Records only the operational
/// <c>view.opened</c> event with the surface slug — never a channel name, URL or secret — so a diagnostics line can
/// never leak user configuration. Thread-safe.
/// </summary>
public sealed class WebhooksPageDiagnostics
{
    private readonly Action<string>? _sink;
    private long _viewsOpened;

    /// <summary>Creates the collector over an optional PII-safe diagnostics sink.</summary>
    public WebhooksPageDiagnostics(Action<string>? sink = null) => _sink = sink;

    /// <summary>The number of <c>view.opened</c> events recorded.</summary>
    public long ViewsOpened => Interlocked.Read(ref _viewsOpened);

    /// <summary>Record that the surface was opened, emitting <c>view.opened slug=WebhooksPage</c>.</summary>
    public void RecordViewOpened()
    {
        Interlocked.Increment(ref _viewsOpened);
        _sink?.Invoke($"view.opened slug={WebhooksPageRegistration.Slug}");
    }
}

File: apps/android/app/src/main/kotlin/com/teslasync/notifications/WebhooksPageModel.kt
// Pure, framework-free metadata + diagnostics for the WebhooksPage notifications surface — the native analogue
// of the cross-cutting concerns the web page owns (web/src/features/notifications/pages/WebhooksPage.tsx, the
// dedicated /notifications/webhooks wrapper that promotes the custom outgoing-webhook channels — HMAC-signed
// payloads + delivery retry policy + delivery audit — from a Settings sub-section to a first-class route). No
// Compose, no Android framework, no HTTP lives here, so the route id + slug are exercised off-device and the
// composable stays a thin render layer. The web page renders no data of its own — it sets the page
// title/subtitle (plus a copy-link affordance) and embeds the shared WebhookChannelsSection component — so this
// surface carries only its navigation identity and the one PII-safe `view.opened` diagnostic, with no page-level
// feed to derive.
//
// `InvalidPackageDeclaration` is suppressed because this surface's mandated directory
// (com/teslasync/notifications — the P3 prompt's allowed-files path) cannot form the package the rest of the
// app's `io.teslasync.android.*` namespace uses, so the package intentionally diverges from the path — exactly as
// the sibling ChannelsPage / ArchivedPage surfaces do. `MatchingDeclarationName` is suppressed for the co-located
// registration + recorder.
@file:Suppress("MatchingDeclarationName", "InvalidPackageDeclaration")

package io.teslasync.android.notifications.webhooks

import io.teslasync.shared.core.diagnostics.Logger

/**
 * Canonical metadata for the WebhooksPage surface. The web page is a top-level notifications route, not a
 * draggable dashboard widget, so there is no web registry row to mirror — this object carries the cross-cutting
 * concerns the surface owes: the navigation [ROUTE_ID] / [WEB_PATH] the host wires (already a metadata-only
 * destination at Destinations.kt) and the diagnostics [SLUG] emitted with the one-shot `view.opened` event
 * (P1/S11). There is no page size or feed metadata because the page renders no data of its own; the embedded
 * WebhookChannelsSection feature view owns the webhook-channel list + the test/signature-preview operations.
 */
object WebhooksPageRegistration {
    /** The navigation destination id (Destinations.kt `page("notificationsWebhooks", "/notifications/webhooks", …)`). */
    const val ROUTE_ID: String = "notificationsWebhooks"

    /** The web route this surface mirrors (deep-link target + the copy-link payload). */
    const val WEB_PATH: String = "/notifications/webhooks"

    /** Diagnostics surface slug emitted with the `view.opened` event (P1/S11). */
    const val SLUG: String = "WebhooksPage"
}

/** Emits the one PII-safe `view.opened` diagnostic with the surface slug (P1/S11); carries no webhook content. */
internal fun recordWebhooksPageOpened(logger: Logger) {
    logger.info(EVENT_VIEW_OPENED, mapOf(FIELD_SURFACE to WebhooksPageRegistration.SLUG))
}

private const val EVENT_VIEW_OPENED = "view.opened"
private const val FIELD_SURFACE = "surface"
